engineering_tools/monitor-ros-cpu.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+"""
+CARMA Platform CPU Monitor Script
+
+Requirements:
+    - Python 3.6 or higher
+    - psutil package (install with: pip3 install psutil)
+
+Usage:
+    1. Open a new terminal before starting CARMA Platform
+    2. Navigate to the directory containing this script:
+       cd /path/to/script/
+
+    3. Run this script:
+       python3 monitor-ros-cpu.py
+
+    4. In a different terminal, start CARMA Platform:
+       ./carma start all
+
+    5. The script will automatically monitor and log CPU/memory usage
+       of all ROS2 nodes and related processes during CARMA operation
+
+    6. To stop monitoring:
+       - Press Ctrl+C in the monitoring terminal
+       - The CSV output file will be saved in the current directory as:
+         'cpu_usage_ros2_nodes.csv' or 'cpu_usage_ros2_nodes-N.csv'
+
+Output:
+    - CSV file containing timestamp, process info, CPU and memory usage
+    - Data can be used to analyze CARMA Platform resource utilization
+"""
+
+import psutil
+import subprocess
+import time
+import csv
+import os
+from datetime import datetime
+
+
+# Function to create a unique filename by appending a number
+def get_unique_filename(base_filename):
```

Comment:
Could we follow the file naming convention of ROS bags and include the date/time? This would make it easier to correlate the script output with the ROS bag file.